### ISSUE TITLE
SNMPV3 Timeout Issue - Resolved

### ISSUE THREAD

**Issue by lextudio-support** [lextudio/sharpsnmplib#297]
I'm having some issues adapting the SNMPv3 GET example given here --> [CodePlex #SNMP Documention](http://sharpsnmplib.codeplex.com/wikipage?title=SNMP%20v3%20Operations&referringTitle=Documentation)

I've taken this code

```
Discovery discovery = Messenger.GetNextDiscovery(SnmpType.GetRequestPdu);
ReportMessage report = discovery.GetResponse(60000, new IPEndPoint(IPAddress.Parse("192.168.1.2"), 161));

var auth = new SHA1AuthenticationProvider(new OctetString("myauthenticationpassword"));
var priv = new DESPrivacyProvider(new OctetString("myprivacypassword"), auth);

GetRequestMessage request = new GetRequestMessage(VersionCode.V3, Messenger.NextMessageId, Messenger.NextRequestId, new OctetString("myname"), new List<variable>{new Variable(new ObjectIdentifier("1.3.6.1.2.1.1.1.0"))}, priv, Messenger.MaxMessageSize, report);
ISnmpMessage reply = request.GetResponse(60000, new IPEndPoint(IPAddress.Parse("192.168.1.2"), 161));
if (reply.Pdu().ErrorStatus.ToInt32() != 0) // != ErrorCode.NoError
{
    throw ErrorException.Create(
        "error in response",
        IPAddress.Parse("192.168.1.2"),
        reply);
}
```

and adapted it to my network configuration and converted it to VB.NET

```
        Dim discovery As Discovery = Messenger.GetNextDiscovery(SnmpType.GetRequestPdu)
        Dim report As ReportMessage = discovery.GetResponse(5000, New IPEndPoint(IPAddress.Parse("10.30.125.40"), 161))

        Dim auth = New SHA1AuthenticationProvider(New OctetString("localadmin"))
        Dim priv = New DESPrivacyProvider(New OctetString("localadmin"), auth)


        Dim request As New GetRequestMessage(VersionCode.V3, _
                                             Messenger.NextMessageId, _
                                             Messenger.NextRequestId, _
                                             New OctetString("localadmin"), _
                                             New List(Of Variable)() From {New Variable(New ObjectIdentifier(".1.3.6.1.4.1.850.100.1.1.3.0"))}, _
                                             priv, _
                                             Messenger.MaxMessageSize, _
                                             report)


        Dim reply As ISnmpMessage

        Try
            reply = request.GetResponse(5000, New IPEndPoint(IPAddress.Parse("10.30.125.40"), 161))

        Catch ex As Exception
            Debug.Print(ex.Message)
        End Try

        If report.Pdu().ErrorStatus.ToInt32() <> 0 Then
            ' != ErrorCode.NoError
            Throw ErrorException.Create("error in response", IPAddress.Parse("10.30.125.40"), reply)
        End If
```

When I run it I get this error on the Reply = request.GetResponse line

```
A first chance exception of type 'Lextm.SharpSnmpLib.Messaging.TimeoutException' occurred in SharpSnmpLib.Full.dll
Exception of type 'Lextm.SharpSnmpLib.Messaging.TimeoutException' was thrown
```

I've confirmed my device is up and running, V3 is working (as well as V2 and V1) using iReasoning so I know I'm doing something wrong in the code but I and stuck... Any thoughts / help would be greatly appreciated.

Thanks!

**Original Reported Date**: 2015-04-17T07:37:17.78-07:00
**Planned For Release**: 
**Original CodePlex ID**: 7271

**Comment by lextudio-support** [comment]
**Copied from CodePlex without authors**:

http://sharpsnmplib.codeplex.com/wikipage?title=600008

Without some network packets it would be pretty difficult to tell what's up. Since you can reproduce the issue easily, you might even debug into the code to see what is the cause.

**Original Posted Date**: 2015-04-19T22:05:20.713-07:00

**Comment by lextudio-support** [comment]
**Copied from CodePlex without authors**:

Hi Lextm - I apologize for not responding sooner - I was out of town and missed your reply. I'll get a wireshark capture later today / tomorrow morning and post it for you here.

**Original Posted Date**: 2015-05-06T08:17:27.237-07:00

**Comment by lextudio-support** [comment]
**Copied from CodePlex without authors**:

Lextm,

I've attached a wireshark capture from my network isolating the capture to only the device i'm communication with. I see that there is a message sent across "privKey Unknown" when I execute request.GetResponse. I've followed your example to the letter (as best I can tell) but maybe I'm missing something..

Thank you for taking a look!

**Original Posted Date**: 2015-05-06T12:05:26.807-07:00

**Comment by lextudio-support** [comment]
**Copied from CodePlex without authors**:

Lextm - I've resolved this issue. Turns out my device is using MD5 Authentication instead of SHA1 Authentication...

**Original Posted Date**: 2015-05-06T13:45:55.773-07:00

**Comment by lextudio-support** [comment]
**Copied from CodePlex without authors**:

Thanks for commenting about the root cause. That should help me reproduce it and analyze whether #SNMP documentation can be updated to list such common issues. Sometimes it is just not easy to get v3 configured as it is kind of complicated.

**Original Posted Date**: 2015-05-06T19:45:07.643-07:00

**Comment by lextudio-support** [comment]
**Original Closed Date**: 2015-05-06T19:45:20.357-07:00
